src/components/poolstats/PoolPositionWidget.tsx

```diff
@@ -39,10 +49,175 @@ const PerformanceCard = styled.div`
 
 export type PoolPositionWidgetProps = {
   poolData: BlendPoolMarkers;
+  offChainPoolStats: OffChainPoolStats | undefined;
 };
 
+type AccountStats = {
+  accountValue: number,
+  accountValue1DAgo: number | null,
+  todaysReturn: number | null,
+  totalReturn: number | null,
+};
+
+type Timestamp = {
+  timestamp: string | number;
+}
+
+function makeEtherscanRequest(
+  fromBlock: number,
+  address: string,
+  topics: (string | null)[],
+  shouldMatchAll: boolean,
+  pageLength = 1000,
+  page?: number,
+  toBlock?: number
+) {
+  let query = `https://api.etherscan.io/api?module=logs&action=getLogs`.concat(
+    `&fromBlock=${fromBlock.toFixed(0)}`,
+    toBlock ? `&toBlock=${toBlock.toFixed(0)}` : '',
+    `&address=${address}`
+  );
+  
+  for (let i = 0; i < topics.length; i += 1) {
+    if (topics[i] === null) continue;
+    query += `&topic${i}=${topics[i]}`;
+
+    if (i === topics.length - 1) break;
+    query += `&topic${i}_${i + 1}_opr=${shouldMatchAll ? 'and' : 'or'}`;
+  }
+
+  if (page) query += `&page=${page}`;
+  query = query.concat(`&offset=${pageLength}`, `&apikey=${process.env.REACT_APP_ETHERSCAN_API_KEY}`);
+
+  return axios.get(query);
+}
+
+function findNearestElementByTime<T extends Timestamp>(arr: T[], timestamp: string | number): T {
+  const costFunc = arr.map((el) => Math.abs(Number(el.timestamp) - Number(timestamp)));
+  const idx = costFunc.reduce((iMin, cost, i, arr) => cost < arr[iMin] ? i : iMin, 0);
+  return arr[idx];
+}
+
 export default function PoolPositionWidget(props: PoolPositionWidgetProps) {
-  // TODO: Incorporate the pool data into the widget
+
+  const { poolData, offChainPoolStats } = props;
+  const { poolStats } = useContext(BlendPoolContext);
+
+  const [{ data: accountData }] = useAccount();
+  const [{ data: accountShareBalance }] = useBalance({
+    addressOrName: accountData?.address,
+    token: poolData.poolAddress,
+    watch: false,
+  });
+
+  const [accountStats, setAccountStats] = useState<AccountStats | null>(null)
+
+  useEffect(() => {
+    let mounted = true;
+
+    if (offChainPoolStats && poolStats && accountShareBalance && accountData) {
+      const tvl = new Big(offChainPoolStats.total_value_locked);
+      const accountValue = tvl.mul(toBig(accountShareBalance.value)).div(poolStats.outstandingShares);
+      const accountAddress = accountData.address.slice(2).toLowerCase();
+
+      const getEvents = makeEtherscanRequest(
+        BLEND_FACTORY_CREATION_BLOCK,
+        poolData.poolAddress,
+        [
+          null,
+          `0x000000000000000000000000${accountAddress}`, // to this user
+          `0x000000000000000000000000${accountAddress}` // from this user
+        ],
+        false // `or` between topics, not `and`
+      );
+
+      const endTime = Date.now() / 1000;
```

Comment:
Nit: wrap the time in _toFixed_ here once rather than each time it is used.

Reply:
Fixed